mic/src/mic/XWSList.java

```diff
@@ -98,6 +98,9 @@ public String getName() {
         }
 
         public String getShip() {
```

Comment:
We should reverse the comparison here to prevent agains null pointers. For example `"yt2400freighter".equals(ship)`